### ISSUE TITLE
Homepage modal: update release logic to make modal visible to a % of users when flipper is enabled

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#12057]
## Description

12/21 new scope request from Dave C: investigate and estimate what would be required to display the homepage modal to a % of users and gradually ramp to 100%, rather than binary off/on flipper. 

This is complicated as the gradual-roll-out logic that comes with flipper pertains to authenticated users only. So: we will need to implement this for _un_authenticated users as well.

The chatbot previously did some similar % release to unauthed users, and our team removed the logic that controlled it. That logic is applicable / reusable for this use case.
* #9332 
* PR for removing that: https://github.com/department-of-veterans-affairs/vets-website/pull/21363/files

**% rollout will begin with 25%.** 
[Release plan ](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/public-websites/home-page/home-page-redesign-initiative/release-plan.md) will document the plan for progression.

### Nice to have: testing, future proofing
1. This will be the 2nd time we've needed a method for % phased rollout to unauthed users. It behooves us to think about how to make this extensible / reusable for future needs. Ryan notes: 
* The previous implementation is not utilizing any sort of hook, which likely has the effect of running the calculation logic every time the render function is called (which is likely a lot more than once.
* A custom hook is probably the way to go, and that'd make this reusable as well.

This issue is time-sensitive, so if making this reusable increases LOE, it can be broken out into a separate ticket.

2. For testing purposes, it would be very helpful to have a means of forcing the modal to appear even if you are not in the 25%.

## Additional context
`va_home_preview_modal` flipper is still available for off/on logic to display the modal at all. 

## Acceptance Criteria
- [ ] The homepage modal is present for 25% of users on page load of VA.gov
  * Decision re: what % to start with: https://dsva.slack.com/archives/C52CL1PKQ/p1671644635556979
- [ ] Document a mechanism for forcing the modal to appear if it is not served on page load (TBD technical feasibility)
- [ ] Google Analytics event is sent when modal is displayed. (Goal: a mechanism to track # of users who see the modal)
  * This AC may be met by #12048 
- [ ] EITHER: Method is written in a reusable way for future simillar needs OR that requirement is broken out into a separate ticket for future work

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [X] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by jilladams** [comment]
Spike notes: https://dsva.slack.com/archives/C52CL1PKQ/p1671643860406759
tl;dr: we can use the chatbot logic. Converting to become an implementation ticket.

**Comment by jilladams** [comment]
Since this was primarily tested locally, and not possible to confirm in a review instance with the given QA steps, I'm moving to PO Review column for Staging verification once it hits staging.

**Comment by jilladams** [comment]
Tested on Staging & verified: 

**Positive: Modal should appear for user assigned `show-homepage-soft-launch-modal` >=25**
- Load http://staging.va.gov
- Check local storage => show-homepage-soft-launch-modal = 13
  - Dismiss, reload: modal doesn't appear
  - Close tab, load in new tab: modal appears

**Negative: modal should _not_ appear for `show-homepage-soft-launch-modal` <25**
- Edit show-homepage-soft-launch-modal and set to 26
  - Reload page: modal doesn't appear
  - Close tab, load in new tab: modal doesn't appear

I then reset the `show-homepage-soft-launch-modal` to 13, reload: modal appears.

Great success. Closing.
